internal/services/eventgrid/eventgrid_partner_namespace_resource.go

```diff
@@ -0,0 +1,331 @@
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tags"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+var _ sdk.ResourceWithUpdate = EventGridPartnerNamespaceResource{}
+
+type EventGridPartnerNamespaceResource struct{}
+
+type EventGridPartnerNamespaceResourceModel struct {
+	PartnerNamespaceName                string                      `tfschema:"name"`
+	Location                            string                      `tfschema:"location"`
+	ResourceGroup                       string                      `tfschema:"resource_group_name"`
+	InboundIPRules                      []PartnerInboundIpRuleModel `tfschema:"inbound_ip_rule"`
+	LocalAuthEnabled                    bool                        `tfschema:"local_auth_enabled"`
+	PartnerRegistrationFullyQualifiedID string                      `tfschema:"partner_registration_id"`
+	PartnerTopicRoutingMode             string                      `tfschema:"partner_topic_routing_mode"`
+	PublicNetworkAccessEnabled          bool                        `tfschema:"public_network_access_enabled"`
+	Endpoint                            string                      `tfschema:"endpoint"`
+	Tags                                map[string]string           `tfschema:"tags"`
+}
+
+type PartnerInboundIpRuleModel struct {
+	IpMask string `tfschema:"ip_mask"`
+	Action string `tfschema:"action"`
+}
+
+func (EventGridPartnerNamespaceResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": &schema.Schema{
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.All(
+				validation.StringIsNotEmpty,
+				validation.StringMatch(
+					regexp.MustCompile("^[-a-zA-Z0-9]{3,50}$"),
+					"EventGrid Partner Namespace name must be 3 - 50 characters long, contain only letters, numbers and hyphens.",
+				),
+			),
+		},
+		"resource_group_name": commonschema.ResourceGroupName(),
+		"location":            commonschema.Location(),
+		"partner_registration_id": &schema.Schema{
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: partnerregistrations.ValidatePartnerRegistrationID,
+		},
+		"inbound_ip_rule": &schema.Schema{
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			MaxItems: 16,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"ip_mask": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.IsCIDR,
+					},
+					"action": {
+						Type:     pluginsdk.TypeString,
+						Optional: true,
+						Default:  string(partnernamespaces.IPActionTypeAllow),
+						ValidateFunc: validation.StringInSlice([]string{
+							string(partnernamespaces.IPActionTypeAllow),
+						}, false),
+					},
+				},
+			},
+		},
+		"local_auth_enabled": &schema.Schema{
```

Comment:
While we've used `auth` in the past, moving forward we should use the unabbreviated form ([ref](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/reference-naming.md?plain=1#L13))

```suggestion
		"local_authentication_enabled": &schema.Schema{
```